src/lib.rs

```diff
@@ -1,4 +1,4 @@
-#![deny(missing_docs)]
+#![warn(missing_docs)]
```

Comment:
Could you revert this change please? Other than that, looks good to me. Thanks!

Reply:
On the second thought, I guess it'll make CI fail anyway. Let's keep it then :)

Reply:
Oh I see. It seems like the new Bevy version breaks CI as its macro generates a struct without documentation.

Reply:
We could `#![allow(missing_docs)]`? See [SO thread](https://users.rust-lang.org/t/solved-applying-attributes-to-macro-included-code/30484/7). Not too sure how to exclude a single derive macro from the missing docs, there's some [forum posts here](https://users.rust-lang.org/t/solved-applying-attributes-to-macro-included-code/30484/7), but it looks messy and might only work for proc macros.

Could allow and wait for bevy to write their docs.